sdk/src/main/java/io/dapr/client/AbstractDaprClient.java

```diff
@@ -110,52 +101,101 @@
-      String appId, String method, HttpExtension httpExtension, Map<String, String> metadata, TypeRef<T> type) {
+  public <T> Mono<T> invokeService(String appId, String method, Object request, HttpExtension httpExtension,
+      Map<String, String> metadata, Class<T> clazz, String contentType) {
+    return this.invokeService(appId, method, request, httpExtension, metadata, TypeRef.get(clazz), contentType);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public <T> Mono<T> invokeService(String appId, String method, HttpExtension httpExtension,
+      Map<String, String> metadata, TypeRef<T> type) {
     return this.invokeService(appId, method, null, httpExtension, metadata, type);
   }
 
   /**
    * {@inheritDoc}
    */
   @Override
-  public <T> Mono<T> invokeService(
-      String appId, String method, HttpExtension httpExtension, Map<String, String> metadata, Class<T> clazz) {
+  public <T> Mono<T> invokeService(String appId, String method, HttpExtension httpExtension,
+      Map<String, String> metadata, TypeRef<T> type, String contentType) {
+    return this.invokeService(appId, method, null, httpExtension, metadata, type, contentType);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public <T> Mono<T> invokeService(String appId, String method, HttpExtension httpExtension,
+      Map<String, String> metadata, Class<T> clazz) {
     return this.invokeService(appId, method, null, httpExtension, metadata, TypeRef.get(clazz));
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public <T> Mono<T> invokeService(String appId, String method, HttpExtension httpExtension,
+      Map<String, String> metadata, Class<T> clazz, String contentType) {
+    return this.invokeService(appId, method, null, httpExtension, metadata, TypeRef.get(clazz), contentType);
+  }
+
   /**
    * {@inheritDoc}
    */
   @Override
   public <T> Mono<T> invokeService(String appId, String method, Object request, HttpExtension httpExtension,
-                                   TypeRef<T> type) {
+      TypeRef<T> type) {
     return this.invokeService(appId, method, request, httpExtension, null, type);
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public <T> Mono<T> invokeService(String appId, String method, Object request,
+      HttpExtension httpExtension, TypeRef<T> type, String contentType) {
+    return this.invokeService(appId, method, request, httpExtension, null, type, contentType);
+  }
+
   /**
    * {@inheritDoc}
    */
   @Override
   public <T> Mono<T> invokeService(String appId, String method, Object request, HttpExtension httpExtension,
-                                   Class<T> clazz) {
+      Class<T> clazz) {
     return this.invokeService(appId, method, request, httpExtension, null, TypeRef.get(clazz));
   }
 
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public <T> Mono<T> invokeService(String appId, String method, Object request,
+      HttpExtension httpExtension, Class<T> clazz, String contentType) {
```

Comment:
contenttype is determined by the serialization of the request object. every call to this method will be serialized to the same type. It feels that contenttype fits better in the constructor for when users pass in a custom serializer. The default serializer will always be application/json.

Another option is to add a new method to the serializer: getContentType(), so there is no overload needed here.

Also, the request object is a place to avoid adding more overloads but I don't think we should add contenttype as a new parameter here anyway.

PS: This comment applies to every method with contenttype param here.

Reply:
Makes sense, I'll make the changes now.